core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -414,6 +415,82 @@ protected DiffRepository getDiffRepos() {
         .check();
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-3170">[CALCITE-3170]
+   * ANTI join on conditions push down generates wrong plan</a>. */
+  @Test public void testCanNotPushAntiJoinConditionsToLeft() {
+    final RelBuilder relBuilder = RelBuilder.create(RelBuilderTest.config().build());
+    // build a rel equivalent to sql:
+    // select * from emp
+    // where emp.empno
```

Comment:
minor: SQL description does not correspond to relBuilder expression:
where emp.**deptno** 
not in (select dept.deptno from dept where emp.deptno > 20)